### PR TITLE
🧹 improve user resource handling in Microsoft 365

### PR DESCRIPTION
**Pull request by chris-rock**
This PR improves the `microsoft.user` resource.

**Simpler init functions**

User can easily created by their object id and principal name now:

```javascript
cnspec> microsoft.user(userPrincipalName: "ben@example.onmicrosoft.com")
microsoft.user: microsoft.user id="a692f8ff-42e3-4156-8ac2-523379d9b664" displayName="Ben Rockwood" userPrincipalName="ben@example.onmicrosoft.com"
```

```javascript
cnspec> microsoft.user(id: "a692f8ff-42e3-4156-8ac2-523379d9b664")
microsoft.user: microsoft.user id="a692f8ff-42e3-4156-8ac2-523379d9b664" displayName="Ben Rockwood" userPrincipalName="ben@example.onmicrosoft.com"
```

**Retrieve owner for application registration**

We added a new `owners` field to `microsoft.application` which allows you to see all owners of your application:

```javascript
cnspec> microsoft.application(name: "chris-testing") { name owners }
microsoft.application: {
  name: "chris-testing"
  owners: [
    0: microsoft.user id="efd9e3d3-eb7a-46f6-a560-09a894e00fdd" displayName="Christoph Hartmann" userPrincipalName="chris@example.onmicrosoft.com"
  ]
}
```

**Group members**

For groups, I it is essential to see all members. This can easily done now via:

```javascript
cnspec> microsoft.groups { name members }
```

### CODE REVIEW

File: providers/ms365/resources/ms365.lr.manifest.yaml [hunk 1]
```diff
@@ -37,6 +37,8 @@ resources:
         min_mondoo_version: 9.0.0
       notes:
         min_mondoo_version: 9.0.0
+      owners:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/ms365/resources/ms365.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/ms365/resources/groups.go [hunk 2]
```diff
@@ -20,7 +17,59 @@ func (m *mqlMicrosoftGroup) id() (string, error) {
 }
 
 func (a *mqlMicrosoftGroup) members() ([]interface{}, error) {
-	return nil, errors.New("not implemented")
+	msResource, err := a.MqlRuntime.CreateResource(a.MqlRuntime, "microsoft", map[string]*llx.RawData{})
+	if err != nil {
+		return nil, err
+	}
+	mqlMicrsoftResource := msResource.(*mqlMicrosoft)
```
Comment (preslavgerchev) [inline providers/ms365/resources/groups.go]:
```suggestion
	mqlMicrosoftResource := msResource.(*mqlMicrosoft)
```
nit

File: providers/ms365/resources/microsoft.go [hunk 3]
```diff
@@ -10,6 +10,41 @@ import (
 	"go.mondoo.com/cnquery/v11/providers/ms365/connection"
 )
 
+type mqlMicrosoftInternal struct {
+	// index users by id
+	idxUsersById map[string]*mqlMicrosoftUser
+	// index users by principal name
+	idxUsersByPrincipalName map[string]*mqlMicrosoftUser
```
Comment (preslavgerchev) [inline providers/ms365/resources/microsoft.go]: nit: it doesnt seem we use this index to fetch users, can we remove it if thats the case?

**Review by preslavgerchev — APPROVED** [review]

File: providers/ms365/resources/microsoft.go (hunk 3 above)
Comment (chris-rock) [reply providers/ms365/resources/microsoft.go]: Yes, lets remove it

**Pull request merged**